### PR TITLE
Update Networking Model from "OrderMetaData" to "MetaData".

### PR DESCRIPTION
**Pull request by hafizrahman**
Part of: #13619

## Description

This is part of an effort to refactor `OrderMetaData` to be `MetaData` as metadata is to be shared with Products, [matching Core where metadata is used for all of post types](https://developer.wordpress.org/plugins/metadata/).

This PR specifically works on the Networking layer first.

## Steps to reproduce

1. The app should build properly,
2. All unit tests should pass,
3. Please smoke test: 
    1. Create a new Order in the web, add custom fields to it, 
    2. Open **Order Details > View Custom Fields** functionality in the app and ensure the custom fields are shown.
    3. Smoke test various things in Order Details.

- [x] This PR includes refactoring; smoke testing of the entire section is needed.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/218">20.0</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Yosemite/Yosemite/Model/Model.swift [hunk 1]
```diff
@@ -74,7 +74,7 @@ public typealias OrderCouponLine = Networking.OrderCouponLine
 public typealias OrderFeeLine = Networking.OrderFeeLine
 public typealias OrderFeeTaxStatus = Networking.OrderFeeTaxStatus
 public typealias OrderGiftCard = Networking.OrderGiftCard
-public typealias OrderMetaData = Networking.OrderMetaData
+public typealias OrderMetaData = Networking.MetaData
```
Comment (hafizrahman) [inline Yosemite/Yosemite/Model/Model.swift:77]: Discussion: 

Given that we're going to be using `Networking.MetaData` also for product, I wonder if it's good to not rename `OrderMetaData` here. Later on when updating Storage and Yosemite, we just add:

`public typealias ProductMetaData = Networking.MetaData`

Some benefits I can think of:

1. Not renaming `Yosemite.OrderMetaData` means not having to update various part of the code that already uses it.
2. Users of Yosemite do not have to care or know that order and product both share the same  `Networking.MetaData` underneath.

Not sure what the downsides are for this approach. What do you think?

Comment (itsmeichigo) [reply Yosemite/Yosemite/Model/Model.swift:77]: This is an interesting take. I was concerned about the initialization of `OrderMetaData` in unit tests, but it looks like the compiler can still accept the initialization of the underlying type. Let's keep this for now and we can change later if we catch any issue with it.

**Review by itsmeichigo — APPROVED** [review]
LGTM :shipit: I did only a quick smoke test around order custom field and attributions and all still look good.

**Pull request merged**